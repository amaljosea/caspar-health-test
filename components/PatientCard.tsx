import Link from "next/link";
import { Patient } from "..";

type PatientCardProps = {
  patient: Patient;
};

export const PatientCard = ({ patient }: PatientCardProps) => {
  return (
    <Link key={patient.patient_id} href={`/detail/${patient.patient_id}`}>
      <li className="container">
        <p>{patient.patient_id}</p>
        <p>
          {patient.first_name} {patient.last_name}
        </p>
        <p>{patient.gender}</p>
        <p>{patient.age}</p>
      </li>
    </Link>
  );
};
