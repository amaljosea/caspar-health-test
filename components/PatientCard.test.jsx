import React from "react";
import { render } from "@testing-library/react";
import { PatientCard } from "./PatientCard";

const testPatient = {
  patient_id: 1,
  first_name: "John",
  last_name: "Doe",
  gender: "male",
  age: 30,
  email: "",
  avatar: "",
};

describe("PatientCard", () => {
  it("renders patient information correctly", () => {
    const { getByText, getByRole } = render(
      <PatientCard patient={testPatient} />
    );

    // Check if patient information is displayed in the component
    expect(getByText("1")).toBeInTheDocument(); // Assuming '1' is the patient_id
    expect(getByText("John Doe")).toBeInTheDocument();

    // Check if the link with the correct href is present
    const link = getByRole("link");
    expect(link).toHaveAttribute("href", "/detail/1"); // Adjust the href as needed
  });
});
