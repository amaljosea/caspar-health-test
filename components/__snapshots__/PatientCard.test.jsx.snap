// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientCard matches snapshot 1`] = `
<DocumentFragment>
  <a
    href="/detail/1"
  >
    <li
      class="container"
    >
      <p>
        1
      </p>
      <p>
        John Doe
      </p>
    </li>
  </a>
</DocumentFragment>
`;
